azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -70,14 +70,16 @@ void setup(final ExecutableFlow flow) {
       // First get the ProjectVersion
       final ProjectVersion projectVersion = getProjectVersion(flow);
 
-      // Setup the project
-      setupProject(projectVersion);
+      synchronized (projectVersion) {
```

Comment:
What's the intention of synchronizing on projectVersion here?

Reply:
updated in comment